File: Maximum Weight Node.java

class Solution{

    public int maxWeightCell(int N, int Edge[]){

       

       int res[]=new int[N];

       for(int i=0;i<N;i++){

           if(Edge[i]!=-1)res[Edge[i]]+=i;

       }

       

       

       int max=0;

       for(int i=0;i<N;i++){

           max=Math.max(max,res[i]);

       }

       

       for(int i=N-1;i>=0;i--){

           if(res[i]==max)return i;

       }

       return 0;

    }

 

}

File: Maximum Weight Node.cpp
class Solution
{
  public:
  int maxWeightCell(int N, vector<int> Edge)
  {
     unordered_map<int,int>mp;
     for(int i=0;i<N;i++){
         mp[Edge[i]]+=i;
     }
     int maxi=0;
     int ans=0;     
     for(auto it:mp){
         if(it.first!=-1 and it.second>=maxi)
         {
             maxi=it.second;
             ans=it.first;
         }         
     }
     return ans;
  }
};
